File: src/main/java/me/nasukhov/TukitaLearner/study/Answer.java
package me.nasukhov.TukitaLearner.study;

import jakarta.persistence.*;

@Entity
@Table(name = "ch_question_replies")
public class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    public String userId;

    @Column(nullable = false)
    public String channelId;

    @ManyToOne
    @JoinColumn(name = "question_id", nullable = false)
    public GroupQuestion question;

    @Column(nullable = false)
    public boolean isCorrect;

    protected Answer() {

    }

    public Answer(String userId, String channelId, GroupQuestion question, boolean isCorrect) {
        this.userId = userId;
        this.channelId = channelId;
        this.question = question;
        this.isCorrect = isCorrect;
    }
}

File: src/main/kotlin/me/nasukhov/TukitaLearner/study/Answer.kt
package me.nasukhov.TukitaLearner.study

import jakarta.persistence.*

@Entity
@Table(name = "ch_question_replies")
class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var id: Long = 0

    @Column(nullable = false, updatable = false)
    lateinit var userId: String
        private set

    @Column(nullable = false, updatable = false)
    lateinit var channelId: String
        private set

    @ManyToOne
    @JoinColumn(name = "question_id", nullable = false)
    lateinit var question: GroupQuestion
        private set

    @Column(nullable = false, updatable = false)
    var isCorrect: Boolean = false
        private set

    // JPA needs this
    protected constructor()

    constructor(userId: String, channelId: String, question: GroupQuestion, isCorrect: Boolean) {
        this.userId = userId
        this.channelId = channelId
        this.question = question
        this.isCorrect = isCorrect
    }
}
